CMakeLists.txt

```diff
@@ -17,10 +17,34 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(add_link_deps Threads::Threads)
 endif()
 
+##############
+# Installation paths
+set(INCLUDE_INSTALL_DIR include/ CACHE PATH "Installation directory for C++ headers")
+set(LIB_INSTALL_DIR lib${LIB_SUFFIX}/ CACHE PATH "Installation directory for libraries")
+
 add_subdirectory(example)
 add_subdirectory(lint)
 
 add_subdirectory(test)
 enable_testing()
 
 install(FILES peglib.h DESTINATION include)
+
+##########
+# Generate -config.cmake file
+# This file will allow to link to this library (peglib.h) with cmake find_package
+
+# Install destination to -config.cmake
+set(INSTALL_DESTINATION_PATH ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}${MSVCARCH_DIR_EXTENSION_EXT})
```

Comment:
`MSVCARCH_DIR_EXTENSION_EXT` looks Windows specific macro. Does this line work on other platforms like macOS and Linux?